VMs/metrics/2_build_baseimage.sh

```diff
@@ -67,6 +67,19 @@ create_initfiles() {
 	sed -i "s/USERNAME/${VM_USERNAME}/g" ./${USER_DATA_NAME}
 	MYARCH=$(go env GOARCH)
 	sed -i "s/ARCH/${MYARCH}/g" ./${USER_DATA_NAME}
+
+	message "Checking new configs"
+	for c in ./"${META_DATA_NAME}" ./"${USER_DATA_NAME}"; do
+		# We use yaml query to check we have a valid set of yaml files
+		# as if you do not (which is pretty easy to do), then cloud-init
+		# is not always verbose with an error, and it can be hard to figure
+		# out why your created VM is not working as expected.
+		# Tested with the yq from https://github.com/mikefarah/yq
+		yq r "${c}" .
```

Comment:
Alas,`yq` -- whilst being a great tool -- doesn't validate, so I'd use `yamllint` here:

- https://github.com/mikefarah/yq/issues/124
- https://github.com/kata-containers/tests/pull/164

Reply:
oh - we might want to review what we are doing [over here](https://github.com/kata-containers/tests/blob/6cebde749343cbf9ef5b059a7a610accc4cbbdad/.ci/lib.sh#L55) as well then. /cc @chavafg 

I tested the yq on my broken yaml, and it bombed with an error and returned 1... looking over on that Issue you have on yq - it does what I need right now - which is to error out on invalid YAML. OK, so it doesn't tell you necessarily why it is invalid .... 

I'll go have a look at yamllint as a potentially more useful alternative then.

Reply:
Well, it seems to detect "almost-valid YAML" (weird indentation), but it accepts pretty much anything else as valid (including an empty file).

Reply:
OK - I don't think I can sensibly make this work with `yamllint` - I don't think the `cloud-init` files are full standard yaml. Here are some examples:

- They have a first line that is required to identify them as cloud-init files, that is not YAML. yamllint complains there is no space after the `#`
```
#cloud-config
# Copyright (c) 2018 Intel Corporation
# 
# SPDX-License-Identifier: Apache-2.0
```

- We have a number of lines that are very long, as they are commandlines - the standard yamllint limit is 80 chars - we could extend that, which sort of defeats the point...
```
- 'bash -c "set -a; source /etc/environment; set +a; curl -L https://dl.google.com/go/go1.10.2.linux-ARCH.tar.gz -o /tmp/go1.10.2.linux-ARCH.tar.gz"'
```

- cloud-init files appear not to have the doc start `---` ( [examples here](http://cloudinit.readthedocs.io/en/latest/topics/examples.html) have none). We can tell yamllint to not fail on that.

- cloud-init write_files sections seem to be the nail in the coffin - they have a `-` in the 0 column in all the examples. If I move that in 2 spaces as you might expect, then yamllint seems to start parsing the contents of the content section, which in the .gitignore case, looks very much like invalid yaml - and thus we fail. I can't see a way around that:
```
-   content: |
        [user]
            name = USERNAME User
            # Safe using a fake email here - we never post to it or expose it outside
            # of the VM.
            email = USERNAME@HOSTNAME
    path: /home/USERNAME/.gitconfig
permissions: '0644'
```

At that point, I'm going to suggest we stick to `yq` for now, as at least it gives us some protection with a cloud-init file that works.